packages/core/src/model.ts

```diff
@@ -50,10 +50,21 @@ export class ChartModel {
 
 		// Loop through all datasets
 		datasets.forEach(dataset => {
-			const group = dataset.label;
-
 			// Update each data point to the new format
 			dataset.data.forEach((datum, i) => {
+				let group;
+
+				if (Tools.getProperty(dataset, "label") === null) {
+					const correspondingLabel = Tools.getProperty(labels, i);
+					if (correspondingLabel) {
+						group = correspondingLabel;
+					} else {
+						group = "Ungrouped";
+					}
+				} else {
+					group = Tools.getProperty(dataset, "label");
+				}
```

Comment:
Just taking the `getProperty` call out?

```suggestion
				const datasetLabel = Tools.getProperty(dataset, "label");
				if (datasetLabel === null) {
					const correspondingLabel = Tools.getProperty(labels, i);
					if (correspondingLabel) {
						group = correspondingLabel;
					} else {
						group = "Ungrouped";
					}
				} else {
					group = datasetLabel;
				}
```